File: clients/java/src/main/java/com/svix/diom/apis/MsgsNamespace.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.ApiException;
import com.svix.diom.HttpClient;
import com.svix.diom.Utils;
import java.io.IOException;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import okhttp3.Headers;
import okhttp3.HttpUrl;
import com.svix.diom.models.MsgNamespaceCreateIn;
import com.svix.diom.models.MsgNamespaceCreateOut;
import com.svix.diom.models.MsgNamespaceGetIn;
import com.svix.diom.models.MsgNamespaceGetOut;

public class MsgsNamespace {
    private final HttpClient client;

    public MsgsNamespace(HttpClient client) {
        this.client = client;
    }

    /** Creates or updates a msgs namespace with the given name. */
    public MsgNamespaceCreateOut create(
        final MsgNamespaceCreateIn msgNamespaceCreateIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/msgs/namespace/create");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            msgNamespaceCreateIn,
            MsgNamespaceCreateOut.class
        );
    }

    /** Gets a msgs namespace by name. */
    public MsgNamespaceGetOut get(
        final MsgNamespaceGetIn msgNamespaceGetIn
    ) throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/msgs/namespace/get");
        return this.client.executeRequest(
            "POST",
            url.build(),
            null,
            msgNamespaceGetIn,
            MsgNamespaceGetOut.class
        );
    }
}

File: clients/python/coyote/apis/msgs_namespace.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    MsgNamespaceCreateIn,
    MsgNamespaceCreateOut,
    MsgNamespaceGetIn,
    MsgNamespaceGetOut,
)


class MsgsNamespaceAsync(ApiBase):
    async def create(
        self,
        msg_namespace_create_in: MsgNamespaceCreateIn,
    ) -> MsgNamespaceCreateOut:
        """Creates or updates a msgs namespace with the given name."""
        body = msg_namespace_create_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/msgs/namespace/create",
            body=body,
            response_type=MsgNamespaceCreateOut,
        )

    async def get(
        self,
        msg_namespace_get_in: MsgNamespaceGetIn,
    ) -> MsgNamespaceGetOut:
        """Gets a msgs namespace by name."""
        body = msg_namespace_get_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/msgs/namespace/get",
            body=body,
            response_type=MsgNamespaceGetOut,
        )


class MsgsNamespace(ApiBase):
    def create(
        self,
        msg_namespace_create_in: MsgNamespaceCreateIn,
    ) -> MsgNamespaceCreateOut:
        """Creates or updates a msgs namespace with the given name."""
        body = msg_namespace_create_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1/msgs/namespace/create",
            body=body,
            response_type=MsgNamespaceCreateOut,
        )

    def get(
        self,
        msg_namespace_get_in: MsgNamespaceGetIn,
    ) -> MsgNamespaceGetOut:
        """Gets a msgs namespace by name."""
        body = msg_namespace_get_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1/msgs/namespace/get",
            body=body,
            response_type=MsgNamespaceGetOut,
        )
